Improve support for Radxa's RK358x-based boards

RadxaNaoki opened the pull request:
Improve support for the following Radxa RK358x-based boards:

- Radxa ROCK 5A
- Radxa ROCK 5B
- Radxa ROCK 5B+ (new)
- Radxa ROCK 5T (new)
- Radxa ROCK 5 ITX/ITX+ (new)
- Radxa ROCK 5C/5C Lite (new)
- Radxa E52C (new)

### Installation

Uncompress the OpenWrt sysupgrade and write it to the micro SD card or eMMC using dd.

Test images can be found [here](https://drive.google.com/drive/folders/1a4g_p03fXWr4vlaEFeK-Zy5MI8u_j3Za?usp=sharing).

### To Do

- [x] Write up commit messages

RadxaNaoki committed:
generic: 6.12: backport gated-fixed-clk driver

Backport gated-fixed-clk driver from Linux v6.13.

This is needed to fix a PCIe controller probe hang on the Radxa ROCK 5
ITX.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
uboot-rockchip: prepare for Rockchip RK3582

Use "rk358x" for RK3588, RK3588S, and upcoming RK3582.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
uboot-rockchip: add support for Rockchip RK3582

Import patches from
 https://patchwork.ozlabs.org/project/uboot/list/?series=468562

Apply patch 3/3 partially due to conflicts.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

1715173329 commented on target/linux/rockchip/patches-6.12/403-01-v6.13-arm64-dts-rockchip-add-and-enable-gpu-node-for-Radxa.patch:
```diff
@@ -0,0 +1,25 @@
+From a98053d098c4ad91a45a3a55604d9574dfc6ffdb Mon Sep 17 00:00:00 2001
```
please rename it to start from `0xx` as it's a backport, the same below.

RadxaNaoki replied:
All backported patches have been renumbered.

1715173329 commented on target/linux/rockchip/patches-6.12/403-02-v6.13-arm64-dts-rockchip-Enable-HDMI0-on-rock-5a.patch:
```diff
@@ -0,0 +1,90 @@
+From f57a8daf6bbd8e71f16693ad6d8421cb881c7fe0 Mon Sep 17 00:00:00 2001
```
why do you add this then revert it in `403-98-arm64-dts-rockchip-Revert-dts-for-Radxa-ROCK-5A.patch`? please remove both

RadxaNaoki replied:
The `hdmi*` nodes have been backported and unnecessary revert patches have been removed.

1715173329 commented on target/linux/rockchip/patches-6.12/404-98-arm64-dts-rockchip-Revert-dts-for-Radxa-ROCK-5B-5BP-5T.patch:
```diff
@@ -0,0 +1,162 @@
+ 		stdout-path = "serial2:1500000n8";
+ 	};
+ 
+-	hdmi0-con {
```
same here

1715173329 commented on target/linux/rockchip/patches-6.12/404-99-arm64-dts-rockchip-Makefile-Add-Radxa-ROCK-5BP-5T.patch:
```diff
@@ -0,0 +1,11 @@
+ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-rock-5b.dtb
+ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-rock-5b-pcie-ep.dtbo
+ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-rock-5b-pcie-srns.dtbo
++dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-rock-5b-plus.dtb
```
why do you make this change separated? it should be merged into the patch where it is added.

RadxaNaoki replied:
Done.

RadxaNaoki committed:
rockchip: backport dts changes for Rockchip RK358x

Backport dts changes up to Linux v6.17 for Rockchip RK358x.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: backport dts changes for Radxa ROCK 5A

Backport dts changes up to Linux v6.17 for Radxa ROCK 5A.

Use power(green) LED instead of heartbeat(blue) LED.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: backport dts changes for Radxa ROCK 5B/5B+/5T

Backport dts changes up to Linux v6.17 for Radxa ROCK 5B, 5B+, and 5T.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: backport dts changes for Radxa ROCK 5 ITX/ITX+

Backport dts changes up to Linux v6.17 for Radxa ROCK 5 ITX and ITX+.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: backport dts changes for Radxa ROCK 5C/5C Lite

Backport dts changes up to Linux v6.17 for Radxa ROCK 5C and 5C Lite.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: backport dts changes for Radxa E52C

Backport dts changes up to Linux v6.17 for Radxa E52C.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: add HATs/M.2 E key module support for Radxa ROCK 5A

Add support for Radxa HATs/M.2 E key module.

- Radxa Dual 2.5G Router HAT (Realtek RTL8125BG/NVMe)
- Radxa Penta SATA HAT (JMicron JMB585)
- Radxa Wireless Module A8 (Realtek RTL8852BE)

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: add M.2 M/E key modules support for Radxa ROCK 5B

Add support for M.2 M/E key modules.

- NVMe
- Radxa Wireless Module A8 (Realtek RTL8852BE)

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki commented:
@1715173329 Thanks for reviewing.
I'm writing the commit messages now, which are TBD.
I'll also add `blkdiscard` to E52C for eMMC/microSD.
This should be my final proposal, and I'm hoping to finish it today.

RadxaNaoki committed:
rockchip: add support for Radxa ROCK 5B+

The Radxa ROCK 5B+ is a single board computer using the Rockchip
RK3588 SoC.

- https://radxa.com/products/rock5/5bp

Hardware
--------

- Quad Cortex-A76 and Quad Cortex-A55 CPU
- Mali-G610 MP4 GPU
- 6 TOPS NPU
- 4/8/16/24/32GB LPDDR5 RAM
- 16/32/64/128/256GB on-board eMMC (optional)
- 16MB SPI NOR flash
- microSD card slot
- Wi-Fi 6 (Realtek RTW8852BE)
- 2.5 Gigabit Ethernet port with PoE (additional PoE HAT required)
- USB 3.1 Gen1 Type-C HOST/OTG port
- 2x USB 3.1 Gen1 Type-A HOST ports
- 2x USB 2.0 Type-A HOST ports
- 2x M.2 M Key connectors with PCIe 3.0 x2
- M.2 B Key connector
- SIM card slot
- RTC battery connector
- PWM fan connector
- 20x2 pin header
- USB Type-C power port

Installation
------------

Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
eMMC using dd.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: add support for Radxa ROCK 5T

The Radxa ROCK 5T is a single board computer using the Rockchip
RK3588(J) SoC.

- https://radxa.com/products/rock5/5t

Hardware
--------

- Quad Cortex-A76 and Quad Cortex-A55 CPU
- Mali-G610 MP4 GPU
- 6 TOPS NPU
- 4/8/16/24/32GB LPDDR5 RAM
- 16/32/64/128/256GB on-board eMMC (optional)
- 16MB SPI NOR flash
- microSD card slot
- Wi-Fi 6E (Intel AX210, limited support)
- 2x 2.5 Gigabit Ethernet ports with PoE (additional PoE module required)
- USB 3.1 Gen1 Type-C HOST/OTG port
- 2x USB 3.1 Gen1 Type-A HOST ports
- 2x USB 2.0 Type-A HOST ports
- 2x M.2 M Key connectors with PCIe 3.0 x2
- M.2 B Key connector
- SIM card slot
- RTC battery connector
- PWM fan connector
- 20x2 pin header
- 5525 12V DC jack

Installation
------------

Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
eMMC using dd.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: add support for Radxa ROCK 5 ITX/ITX+

The Radxa ROCK 5 ITX(+) is a Mini-ITX form factor computer using the
Rockchip RK3588 SoC.

- https://radxa.com/products/rock5/5itx
- https://radxa.com/products/rock5/5itxp

Hardware
--------

- Quad Cortex-A76 and Quad Cortex-A55 CPU
- Mali-G610 MP4 GPU
- 6 TOPS NPU
- 4/8/16/24/32GB LPDDR5 RAM
- on-board eMMC
- 16MB SPI NOR flash
- microSD card slot
- 2x 2.5 Gigabit Ethernet ports with PoE (additional PoE module required)
- USB 3.1 Gen1 Type-C HOST/OTG port
- 4x USB 3.1 Gen1 Type-A HOST ports
- 2x USB 2.0 Type-A HOST ports
- M.2 M Key connector with PCIe 3.0 x2
- 4x SATA connectors (ITX only)
- 2nd M.2 M Key connector with PCIe 3.0 x2 (ITX+ only)
- M.2 E Key connector with PCIe 2.1 x1 and USB 2.0
- RTC battery socket for CR1220
- 4pin PWM fan connector
- Serial console pin header
- Front panel pin headers
- 24pin ATX power connector
- 5525 12V DC jack

Installation
------------

Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
eMMC using dd.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: add support for Radxa ROCK 5C/5C Lite

The Radxa ROCK 5C (Lite) is a single board computer using the Rockchip
RK3588S2 (RK3582) SoC.

- https://radxa.com/products/rock5/5c

Hardware
--------

- Quad (Dual) Cortex-A76 and Quad Cortex-A55 CPU
- Mali-G610 MP4 GPU (5C only)
- 6 (5) TOPS NPU
- 1/2/4/8/16/32GB LPDDR4X RAM
- eMMC/SPI NOR flash connector
- microSD card slot
- Wi-Fi 6 (AIC8800D80, not yet supported)
- Gigabit Ethernet port with PoE (additional PoE HAT required)
- USB 3.1 Gen1 Type-A HOST/OTG port
- USB 3.1 Gen1 Type-A HOST port
- 2x USB 2.0 Type-A HOST ports
- FPC connector with PCIe 2.1 x1
- PWM fan connector
- 20x2 pin header
- USB Type-C power port

Installation
------------

Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
eMMC using dd.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki committed:
rockchip: add support for Radxa E52C

The Radxa E52C is a compact network computer using the Rockchip RK3582
SoC.

- https://radxa.com/products/network-computer/e52c

Hardware
--------

- Dual Cortex-A76 and Quad Cortex-A55 CPU
- 5 TOPS NPU
- 2/4/8GB LPDDR4 RAM
- 16/32/64GB on-board eMMC
- microSD card slot
- 2x 2.5 Gigabit Ethernet ports
- USB 3.1 Gen1 Type-A HOST/OTG port
- USB Type-C debug port
- USB Type-C power port

Installation
------------

Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
eMMC using dd.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/19867
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

wurzerj reviewed (approved):
The changes look good to me. Created a custom build adding this patch on top of master and tested on Radxa E52C. It boots and so far seems to work stable. Big thank you!
